driver/ppm_tp.h

```diff
@@ -20,4 +20,65 @@ typedef enum {
 } ppm_tp_code;
 
 extern const char *tp_names[];
-extern ppm_tp_code tp_from_name(const char *tp_path);
\ No newline at end of file
+
+#ifndef __KERNEL__
+
+#include <stdbool.h>
+
+typedef struct
+{
+	bool tp[TP_VAL_MAX];
+} interesting_ppm_tp_set;
+
+void tp_set_from_sc_set(const bool *sc_set, bool *tp_set);
+
+#ifdef SCAP_HANDLE_T
+
+static int handle_ppm_sc_mask(SCAP_HANDLE_T *handle, bool *sc_set, bool enable, unsigned int ppm_sc,
```

Comment:
To be honest, I don't like too much the callback-approach, btw we can start to merge to have a first working solution and then we can try to refactor it a little bit! I'm already working on a partial refactor at least in the modern bpf https://github.com/Andreagit97/libs/commit/2057b29bc2125870bd53eb43a3f69a67ce4bd2f8 :)